src/precice/impl/ParticipantImpl.cpp

```diff
@@ -787,10 +786,10 @@ void ParticipantImpl::setMeshQuad(
 {
   PRECICE_TRACE(meshName, firstVertexID,
                 secondVertexID, thirdVertexID, fourthVertexID);
-  PRECICE_CHECK(_dimensions == 3, "setMeshQuad is only possible for 3D cases."
-                                  " Please set the dimension to 3 in the preCICE configuration file.");
   PRECICE_REQUIRE_MESH_MODIFY(meshName);
   MeshContext &context = _accessor->usedMeshContext(meshName);
+  PRECICE_CHECK(context.mesh->getDimensions() == 3, "setMeshQuad is only possible for 3D cases."
+                                                    " Please set the dimension to 3 in the preCICE configuration file.");
```

Comment:
```suggestion
  PRECICE_CHECK(context.mesh->getDimensions() == 3, "setMeshQuad is only possible for 3D meshes."
                                                    " Please set the mesh dimension to 3 in the preCICE configuration file.");
```
More occasions below